atlasdb-commons/src/main/java/com/palantir/exception/NotInitializedException.java

```diff
@@ -16,18 +16,39 @@
 package com.palantir.exception;
 
 import com.palantir.common.exception.AtlasDbDependencyException;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
 import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+
+public class NotInitializedException extends AtlasDbDependencyException implements SafeLoggable {
+
+    private static final String EXCEPTION_MESSAGE =
+            "The object is not yet initialized, and as a result, all calls to it will fail. Interactions with this"
+                    + " object should not take place until after initialization has completed. For example, if your"
+                    + " service that handles REST requests relies on this object, it should not service REST requests"
+                    + " until initialization has completed, as otherwise all calls are guaranteed to fail. It is"
+                    + " expected that initialization of some object types do take some time.";
+    private final String objectNotInitialized;
 
-public class NotInitializedException extends AtlasDbDependencyException {
     public NotInitializedException(String objectNotInitialized) {
-        super(exceptionMessage(objectNotInitialized));
+        super(EXCEPTION_MESSAGE);
+        this.objectNotInitialized = objectNotInitialized;
     }
 
     public NotInitializedException(String objectNotInitialized, Throwable throwable) {
-        super(exceptionMessage(objectNotInitialized), throwable);
+        super(EXCEPTION_MESSAGE, throwable);
+        this.objectNotInitialized = objectNotInitialized;
+    }
+
+    @Override
+    public @Safe String getLogMessage() {
```

Comment:
nit: I believe it's more idiomatic to have the @Safe annotation on the return type be above the function, like with @Override.